src/actions/ISoilAction.C

```diff
@@ -37,6 +37,7 @@ ISoilAction::validParams()
       "layer_variable", "The auxvariable providing the soil layer identification.");
   params.addRequiredParam<std::vector<unsigned int>>(
       "layer_ids",
+      {},
```

Comment:
I don't think we need a default for a required parameter. However, it looks like line 107 (`backbone_curve_files`) needs a default..

Reply:
Actually, there are quite a few other addParam<std::vector<Real>> lines with missing default values after line 107. Do they need a default too, or am I missing something?

Reply:
I add the default value mainly based on the reported test errors. Sometime if there is isParamValid() check later, I'll just skip adding default values. Let me double check, the json failure may cover some actual failed tests.

Reply:
I've removed the default value for the required parameter. As the parameters after line 107 you mentioned above, there are comments like "params required for user_defined backbone curve: soil_type = 'user_defined", we prefer to throw out an error message rather than give a default one to remind users to set the parameter when they use e.g. soil_type = 'user_defined'. I run the tests locally with the new changes in MOOSE, all the tests pass
```
  // params required for user_defined backbone curve: soil_type = 'user_defined'
  params.addParam<std::vector<FileName>>(
      "backbone_curve_files",
      "The vector of file names of the files containing "
      "stress-strain backbone curves for the different soil layers. The "
      "size of the vector should be same as the size of layer_ids. All files "
      "should contain the same number of stress-strain points. Headers are not "
      "expected and it is assumed that the first column corresponds to strain values "
      "and the second column corresponds to the stress values. Additionally, two "
      "segments of a backbone curve cannot have the same slope.");
```